mesonbuild/dependencies/configtool.py

```diff
@@ -78,35 +78,11 @@ def find_config(self, versions: T.List[str], returncode: int = 0) \
         for potential_bin in find_external_program(
                 self.env, self.for_machine, self.tool_name,
                 self.tool_name, self.tools, allow_default_for_cross=self.allow_default_for_cross):
-            if not potential_bin.found():
+            tool, out = self._check_config(potential_bin, [], versions, returncode)
+            if tool is None and out is None:
                 continue
-            tool = potential_bin.get_command()
```

Comment:
There's a lot of churn here, but it's just refactoring to the _check_config function. Open to ideas to make this churn less.